### PR TITLE
Prevent adding extra `Snippets` path component when converting to symbol graph symbols

### PR DESCRIPTION
**Pull request by bitjammer**
rdar://104044227

- [x] Added tests
- [x] Ran the `./bin/test` script and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

**Comment by bitjammer** [comment]
@swift-ci Please test

File: Sources/snippet-extract/Utility/SymbolGraph+Snippet.swift [hunk 1]
```diff
@@ -19,14 +19,18 @@ extension SymbolGraph.Symbol {
...
+        var pathComponents = Array(snippet.sourceFile.absoluteURL.deletingPathExtension().pathComponents[...])
 
         guard let snippetsPathComponentIndex = pathComponents.firstIndex(where: {
             $0 == "Snippets"
         }) else {
             throw SnippetExtractCommand.ArgumentError.snippetNotContainedInSnippetsDirectory(snippet.sourceFile)
         }
-        pathComponents = pathComponents[snippetsPathComponentIndex...]
+
+        // In theory, there may be differently named snippet root directories in the future.
```
Comment (ethan-kusters) [inline Sources/snippet-extract/Utility/SymbolGraph+Snippet.swift:30]: Should we have a test case for this possibility too?

**Review by ethan-kusters — APPROVED** [review]

File: Sources/snippet-extract/Utility/SymbolGraph+Snippet.swift (hunk 1 above)
Comment (bitjammer) [reply Sources/snippet-extract/Utility/SymbolGraph+Snippet.swift:30]: It's not *currently possible* even though the URL is actually completely arbitrary right now. There is already a check for a `Snippets` sub-root with a hard error in snippet-extract.

**Comment by bitjammer** [comment]
@swift-ci Please test

**Comment by bitjammer** [comment]
@swift-ci Please test

**Pull request merged**